File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
	public static String longestCommonPrefix(String[] strs) {

		int length = strs[0].length();
		String firstString = strs[0];
		String prefix = "";
        
        if(strs.length == 1){
            return strs[0];
        }

		for(int j=0; j < length; j++){
			String str = firstString.substring(0, j+1);
			for(int i=1; i < strs.length; i++) {
				if(!strs[i].startsWith(str)){
					break;
				}

				if(strs[i].startsWith(str)
					&& i == strs.length - 1){
					prefix = str;
				}
			}
		}

		return prefix;
	}
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:

    def longestCommonPrefix(self, strs: List[str]) -> str:

        if len(strs) == 1:
            return strs[0]

        prefix = ""

        for i in range(len(strs[0])):
            prev = strs[0][:i+1]
            for j in range(1, len(strs)):
                if not strs[j].startswith(prev):
                    break

                if strs[j].startswith(prev) and j == len(strs) - 1:
                    prefix = prev

        return prefix 